src/shared/services/bookmarks-views-plays-service.ts

```diff
@@ -0,0 +1,329 @@
+
+export class BookmarksViewsPlaysService {
+	public static async action(
+		action: EventAction,
+		contentType: EventContentType,
+		contentUuid: string,
+		user?: Avo.User.User,
+		silent: boolean = true
+	): Promise<void> {
+		try {
+			// bundle is handled the same way as a collection
+			const contentTypeSimplified = contentType === 'bundle' ? 'collection' : contentType;
+
+			const mutation = get(EVENT_QUERIES, [action, contentTypeSimplified, 'query']);
+			const variables = get(
+				EVENT_QUERIES,
+				[action, contentTypeSimplified, 'variables'],
+				() => {}
+			)(contentUuid, get(user, 'profile.id'));
+			if (!mutation || !variables) {
+				throw new CustomError('Failed to find query/variables in query lookup table');
+			}
+
+			const response = await dataService.mutate({
+				mutation,
+				variables,
+				update: ApolloCacheManager.clearBookmarksViewsPlays,
+			});
+
+			if (response.errors) {
+				throw new CustomError('Graphql errors', null, { errors: response.errors });
+			}
+
+			if (
+				response.data &&
+				get(response, ['data', Object.keys(response.data)[0], 'affected_rows']) === 0
+			) {
+				if (action === 'play' || action === 'view') {
+					// No row exists yet that can be increased, we should add a row with count = 1
+					this.insertInitialCount(
+						`${action}Init` as EventInitAction,
+						contentType,
+						contentUuid,
+						user,
+						silent
+					);
+				} else {
+					// Bookmark related action failed
+					console.error('Failed to store action into the database', null, {
+						response,
+						variables,
+					});
+					toastService.danger(
+						action === 'bookmark'
+							? i18n.t('Het aanmaken van de bladwijzer is mislukt')
+							: i18n.t('Het verwijderen van de bladwijzer is mislukt')
+					);
+				}
+			}
+
+			// Finished incrementing
+		} catch (err) {
+			const error = new CustomError('Failed to store user action to the database', err, {
+				action,
+				contentType,
+				contentUuid,
+				user,
+			});
+			if (silent) {
+				console.error(error);
+			} else {
+				throw error;
+			}
+		}
+	}
+
+	public static async getItemCounts(
+		itemUuid: string,
+		user: Avo.User.User | null
+	): Promise<BookmarkViewPlayCounts> {
+		const response = await dataService.query({
+			query: GET_ITEM_BOOKMARK_VIEW_PLAY_COUNTS,
+			variables: { itemUuid, profileId: get(user, 'profile.id', null) },
+		});
+		const isBookmarked = !!get(response, 'data.app_item_bookmarks[0]');
+		const bookmarkCount = get(response, 'data.app_item_bookmarks_aggregate.aggregate.count', 0);
+		const viewCount = get(response, 'data.app_item_views[0].count', 0);
+		const playCount = get(response, 'data.app_item_plays[0].count', 0);
+		return {
+			bookmarkCount,
+			viewCount,
+			playCount,
+			isBookmarked,
+		};
+	}
+
+	public static async getCollectionCounts(
+		collectionUuid: string,
+		user: Avo.User.User | null
+	): Promise<BookmarkViewPlayCounts> {
+		const response = await dataService.query({
```

Comment:
Idem?

Reply:
die exceptie wordt opgevangen in collectiondetail